torchgeo/datasets/usavars.py

```diff
@@ -0,0 +1,242 @@
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.labels = labels
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        self.files = self._load_files()
+
+        self.label_dfs = dict(
+            [
+                (
+                    lab,
+                    pd.read_csv(os.path.join(self.root, lab + ".csv"), index_col="ID"),
+                )
+                for lab in self.labels
+            ]
+        )
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        tif_file = self.files[index]
+        id_ = tif_file[5:-4]
+
+        sample = {
+            "labels": Tensor(
+                [self.label_dfs[lab].loc[id_][lab] for lab in self.labels]
+            ),
+            "image": self._load_image(os.path.join(self.root, "uar", tif_file)),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self) -> List[str]:
+        file_path = os.path.join(self.root, "uar")
+        files = os.listdir(file_path)
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        with rasterio.open(path) as f:
+            array: "np.typing.NDArray[np.int_]" = f.read()
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, "uar")
+        csv_pathname = os.path.join(self.root, "*.csv")
+
+        # if glob.glob(pathname) and len(glob.glob(csv_pathname)) == 7:
```

Comment:
Because we download all the CSVs shouldn't we be checking `len(glob.glob(csv_pathname)) == 7`?